response/page_path_test.go

```diff
@@ -0,0 +1,265 @@
+package response
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+type TestCase struct {
+	uri      string
+	expected string
+}
+
+func TestGetPagePath(t *testing.T) {
+	Convey("Given a list of URIs", t, func() {
+		ctx := context.Background()
+
+		visualisationsEndpoints := []TestCase{
+			{
+				uri:      "/visualisations/dvc1945/seasonalflu/index.html",
+				expected: "/visualisations/dvc1945",
+			},
+			{
+				uri:      "/visualisations/dvc1945/duetoinvolving/fallback.png",
+				expected: "/visualisations/dvc1945",
+			},
+		}
+		resourceEndPointPaths := []TestCase{
+			{
+				uri:      "/generator?uri=/economy/economicoutputandproductivity/output/bulletins/economicactivityandsocialchangeintheukrealtimeindicators/15february2024/426e63a0&format=csv",
+				expected: "/economy/economicoutputandproductivity/output/bulletins/economicactivityandsocialchangeintheukrealtimeindicators/15february2024",
+			},
+			{
+				uri:      "/chartimage?uri=/economy/economicoutputandproductivity/output/bulletins/economicactivityandsocialchangeintheukrealtimeindicators/15february2024/426e63a0",
+				expected: "/economy/economicoutputandproductivity/output/bulletins/economicactivityandsocialchangeintheukrealtimeindicators/15february2024",
+			},
+			{
+				uri:      "/file?uri=/aboutus/whatwedo/programmesandprojects/sustainabledevelopmentgoals/c3ac8759.png",
+				expected: "/aboutus/whatwedo/programmesandprojects/sustainabledevelopmentgoals",
+			},
+			{
+				uri:      "/generator?format=csv&uri=/economy/grossvalueaddedgva/timeseries/abml/pn2/previous/v29",
+				expected: "/economy/grossvalueaddedgva/timeseries/abml/pn2",
+			},
+		}
+		resourceEndPointPathsWithEncodedURI := []TestCase{
+			{
+				uri:      "/generator?uri=%2Feconomy%2Feconomicoutputandproductivity%2Foutput%2Fbulletins%2Feconomicactivityandsocialchangeintheukrealtimeindicators%2F15february2024%2F426e63a0%26format%3Dcsv",
```

Comment:
This one does not look right, as it seems like the & from the `format` query param has also been encoded. It's only the value for the `uri` path param that should be encoded. This is not a real use case. It's still valuable, so if you remove the encoding from anything that's not the value of the `uri` param, then that should make it look like a real use case.